x-pack/plugins/apm/public/store/reactReduxRequest/traceList.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { Request } from 'react-redux-request';
+import { createSelector } from 'reselect';
+import mockTraceList from '../mockData/mockTraceList.json';
+import { createInitialDataSelector } from './helpers';
+
+const ID = 'traceList';
+const INITIAL_DATA = [];
+const withInitialData = createInitialDataSelector(INITIAL_DATA);
+
+const selectRRR = (state = {}) => state.reactReduxRequest;
+
+export const selectTraceList = createSelector(
+  [selectRRR],
+  reactReduxRequest => {
+    return withInitialData(reactReduxRequest[ID]);
+  }
+);
```

Comment:
Are there any advantages to using re-select instead of:

```js
export function selectTraceList(state) {
  return withInitialData(state.reactReduxRequest[ID]);
}
```

Reply:
Honestly don't remember now. I may have not realized `withInitialData` was doing a memoized reselect under the hood.